text/0000-deprecate-globals-resolver.md

```diff
@@ -0,0 +1,115 @@
+- Start Date: 2018-05-08
+- RFC PR: (leave this empty)
+- Ember Issue: (leave this empty)
+
+# Summary
+
+Deprecate all use of:
+
+- Ember Globals Resolver (looks up a class via a global namespace such as "App")
+- Creation of a Global Namespace (`var App = Ember.Namespace.create();`)
+- Ember.TEMPLATES array
+- &lt;script type="text/handlebars" data-template-name="path/to/template"&gt;
+
+Use of any of the above should trigger a deprecation warning, with a target
+of version 4.0
+
+# Motivation
+
+Over the past years we have transitioned to using Ember-CLI as the main way
+to compile Ember apps. The globals resolver is a holdover and primarily
+facilitates use of Ember without Ember-CLI.
+
+# The Globals Resolver
+
+For those who are not aware, the globals resolver is available via `@ember/globals-resolver` or
+`Ember.DefaultResolver`. For more information, see the
+[api](https://www.emberjs.com/api/ember/release/classes/GlobalsResolver/properties).
+Using it looks like the following:
+
+```js
+// app.js
+var App = Ember.Application.create();
+
+App.Router.map(function() {
+  this.route('about');
+});
+
+App.AboutRoute = Ember.Route.extend({
+  model: function() {
+    return ['red', 'yellow', 'blue'];
+  }
+});
+```
+
+```html
+// index.html
+<script type="text/x-handlebars" data-template-name="about">
+  <ul>
+    {{#each model as |item|}}
+      <li>{{item}}</li>
+    {{/each}}
+  </ul>
+</script>
+```
+
+# Transition Path
+
+Primarily, the transition path is to recommend using Ember-CLI.
```

Comment:
One small clarifying point for this transition path, is that ember-cli's own default resolver [ember-resolver](https://github.com/ember-cli/ember-resolver) currently still extends from the globals resolver. In order to implement this RFC, and deprecate the globals resolver we need to change the ember-cli resolver so that it does _not_ extend from the globals resolver (otherwise even ember-cli users would get a deprecation).

Reply:
Comment added